tests/NHSD.BuyingCatalogue.Organisations.Api.UnitTests/Builders/OrganisationControllerBuilder.cs

```diff
@@ -53,29 +53,29 @@ internal OrganisationControllerBuilder WithUpdateOrganisation(Organisation organ
             return this;
         }
 
-        internal OrganisationControllerBuilder WithCreateOrganisation()
-        {
-            var repositoryMock = new Mock<IOrganisationRepository>();
-            repositoryMock.Setup(x => x.CreateOrganisationAsync(It.IsAny<Organisation>()));
-            _organisationRepository = repositoryMock.Object;
-            return this;
-        }
-
         public OrganisationControllerBuilder WithOrganisationRepository(IOrganisationRepository organisationRepository)
         {
             _organisationRepository = organisationRepository;
             return this;
         }
 
-        public OrganisationControllerBuilder WithCreateOrganisationService(bool isSuccess, string resultValue)
+        public OrganisationControllerBuilder WithCreateOrganisationServiceReturningResult(bool isSuccess, string resultValue)
         {
+            WithCreateOrganisation();
             var createOrganisationService = new Mock<ICreateOrganisationService>();
             createOrganisationService.Setup(s => s.CreateAsync(It.IsAny<CreateOrganisationRequest>()))
-                .ReturnsAsync(isSuccess ? Result.Success(resultValue) : Result.Failure<string>(new ErrorDetails(resultValue)));
+                .ReturnsAsync(isSuccess ? Result.Success(Guid.Parse(resultValue)) : Result.Failure<Guid>(new ErrorDetails(resultValue)));
             _createOrganisationService = createOrganisationService.Object;
             return this;
         }
```

Comment:
I was just thinking about this and I think I'd be inclined to refactor this slightly as I think it's best to avoid logic in tests and test setup methods wherever possible. I think I'd probably do something like this:

```c#
internal OrganisationControllerBuilder WithCreateOrganisationServiceReturningResult(Guid result)
{
	return WithCreateOrganisationServiceReturningResult(Result.Success(result));
}

internal OrganisationControllerBuilder WithCreateOrganisationServiceReturningResult(string result)
{
	return WithCreateOrganisationServiceReturningResult(Result.Failure<Guid>(new ErrorDetails(result)));
}

private OrganisationControllerBuilder WithCreateOrganisationServiceReturningResult(Result<Guid> result)
{
	WithCreateOrganisation();
	var createOrganisationService = new Mock<ICreateOrganisationService>();
	createOrganisationService.Setup(s => s.CreateAsync(It.IsAny<CreateOrganisationRequest>()))
		.ReturnsAsync(result);
	_createOrganisationService = createOrganisationService.Object;
	return this;
}
```
What do you think?

(Again, it's not that important but the `With` methods you've added can all be `internal`.)

Reply:
Love it. 

I tend to have the habit of leaving things more open - leaving users options if you will. 
After a conversation with @sam-roberts-bjss on similar topic, I understand that this not preferable in our setting.

I will try to keep in this mind going forward, thank you for revisiting this!